src/main/java/com/springdata/config/ApplicationConfig.java

```diff
@@ -0,0 +1,79 @@
+package com.springdata.config;
+
+import org.hibernate.jpa.HibernatePersistenceProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
+
+import java.util.Properties;
+
+@Configuration
+@ComponentScan("com.springdata")
+@EnableJpaRepositories("com.springdata.repository")
+@PropertySource("classpath:application.properties")
+public class ApplicationConfig {
+    @Value("${url}")
+    private String url;
+    @Value("${user}")
+    private String user;
+    @Value("${driver}")
+    private String driver;
+    @Value("${password}")
+    private String password;
+    @Value("${hibernate.dialect}")
+    private String dialect;
+    @Value("${ddl-auto}")
+    private String ddlAuto;
+    @Value("${hibernate.show_sql}")
+    private String showSQL;
+
+    @Bean
+    public DriverManagerDataSource DataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
```

Comment:
Please use overload constructor 
![image](https://user-images.githubusercontent.com/20267148/57582163-ffc72300-74c9-11e9-97ab-380b7a3e40a7.png)
It would make your code better.